### PR TITLE
PAINTROID-758 Implement LineTool specific undo/redo behaviour

### PR DESCRIPTION
**Pull request by Lenkomotive**
[PAINTROID-758](https://catrobat.atlassian.net/browse/PAINTROID-758)

## New Features and Enhancements
- rebuild line sequence on undo/redo for linetool, user can reactive the line and alter it
## Refactorings and Bug Fixes
- Refactorings and Bug Fixes

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Add the link to the ticket in Jira in the description of the PR
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Make sure to organize imports with 'make sort' before committing
- [x] Confirm that the changes follow the project’s coding guidelines (Wiki)
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Add new information to the [Wiki](https://github.com/Catrobat/Paintroid-Flutter/wiki)

[PAINTROID-758]: https://catrobat.atlassian.net/browse/PAINTROID-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bakicelebi — APPROVED** [review]
👍🏻

**Review by bakicelebi — CHANGES_REQUESTED** [review]
Fix conflicts.

File: lib/core/tools/line_tool/line_tool.dart [hunk 1]
```diff
@@ -312,4 +354,20 @@ class LineTool extends Tool with EquatableMixin {
 
     return outsidePoint;
   }
+
+  void _rebuildVertexStack() {
+    reset();
```
Comment (bakicelebi) [inline lib/core/tools/line_tool/line_tool.dart]: Can this method be named more clearly? Not clear what it is resetting.

File: test/utils/ui_interaction.dart [hunk 2]
```diff
@@ -134,4 +135,11 @@ class UIInteraction {
     await tester.tapAt(position);
     await tester.pumpAndSettle();
   }
+
+  static void expectVertexStackLength(int length) {
+    final container =
+        ProviderScope.containerOf(tester.element(find.byType(App)));
+    final tool = container.read(toolBoxStateProvider).currentTool;
```
Comment (bakicelebi) [inline test/utils/ui_interaction.dart]: getCurrentTool can be called here?

Comment (Lenkomotive) [reply test/utils/ui_interaction.dart]: makes sense 👍🏼

**Review by bakicelebi — APPROVED** [review]

**Pull request merged**